src/main/java/com/salesforce/dataloader/security/EncryptionAesUtil.java

```diff
@@ -0,0 +1,198 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.dataloader.security;
+
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.GeneralSecurityException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.InvalidParameterSpecException;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * com.salesforce.dataloader.security
+ *
+ * @author xbian
+ */
+public class EncryptionAesUtil {
+
+    private static final Logger LOGGER = Logger.getLogger(EncryptionAesUtil.class);
+
+    // Support single text encryption and decryption
+
+    static public Cipher cipher;
+    static public byte[] cipherKey;
+
+
+    public static final int ENCRYPTION_KEY_LENGTH_IN_BYTES = 16;
+    public static final String DEFAULT_KEYFILE_NAME = "dataLoader.key";
+
+    static {
+        try {
+            cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+
+        } catch (Exception e) {
+            LOGGER.error("Error to initialize Encryption: " + e.getMessage());
+            throw new  RuntimeException("Error to initialize Encryption: ", e);
+        }
+    }
+
+
+    private byte[] extractIvBytes(byte[] cipheredText) {
+        byte[] iv = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        System.arraycopy(cipheredText, 0, iv, 0, ENCRYPTION_KEY_LENGTH_IN_BYTES);
+        return iv;
+    }
+
+    private byte[] extractCipherContent(byte[] cipheredText) {
+        byte[] cipheredPassword = new byte[cipheredText.length - ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        System.arraycopy(cipheredText, ENCRYPTION_KEY_LENGTH_IN_BYTES, cipheredPassword, 0, cipheredText.length - ENCRYPTION_KEY_LENGTH_IN_BYTES);
+        return cipheredPassword;
+
+    }
+
+    private byte[] concatenateByteArray(byte[] a, byte[] b) {
+
+        byte[] c = new byte[a.length + b.length];
+        System.arraycopy(a, 0, c, 0, a.length);
+        System.arraycopy(b, 0, c, a.length, b.length);
+        return c;
+    }
+
+    public byte[] generateEncryptionKey() {
+        // The secure random uses threadId and tick to make sure it is not repeatable
+        SecureRandom random = new SecureRandom();
+        byte bytes[] = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        random.nextBytes(bytes);
+        return bytes;
+    }
+
+    public String createUserProfileKeyName() {
+        String path = System.getProperty("user.home") + File.separator + "dataloader";
```

Comment:
`Paths.get(System.getProperty("user.home"), "dataloader").toString()` is what you want. 
Also, the path returned is not consistent with other config files path but I can fix this in my world-accessible file bug(W-4648801)